p2p/secret_connection.go

```diff
@@ -298,7 +298,7 @@ func shareAuthSignature(sc *SecretConnection, pubKey crypto.PubKeyEd25519, signa
 // sha256
 func hash32(input []byte) (res *[32]byte) {
 	hasher := sha256.New()
-	hasher.Write(input) // does not error
+	_, _ = hasher.Write(input) // ignoring error
```

Comment:
change comment back

Reply:
`// does not error` is false though? this function throws an error and we don't even acknowledge it.

Reply:
@zramsay conceptually you are right since it implements io.Writer. However, the hash library has a contract that it never throws an error, please see https://golang.org/pkg/hash/#Hash
or 
<img width="971" alt="screen shot 2017-10-03 at 4 27 59 pm" src="https://user-images.githubusercontent.com/4898263/31152113-da796242-a857-11e7-9d4d-46c80279a613.png">

Reply:
ah, gotcha, thanks